src/Silverback.Integration.Kafka/Messaging/Broker/KafkaConsumer.cs

```diff
@@ -127,7 +127,8 @@ internal void OnPartitionsRevoked()
 
             AsyncHelper.RunSynchronously(WaitUntilChannelsManagerStopsAsync);
 
-            if (!Endpoint.Configuration.IsAutoCommitEnabled)
+            if (!Endpoint.Configuration.IsAutoCommitEnabled &&
+                !Endpoint.Configuration.GroupId.Equals(ConfluentConsumerBuilder.UnusedGroupId, StringComparison.Ordinal))
```

Comment:
This is the same as simply doing `string1 == string2`.
I'd add a boolean to the `ConsumerConfig` like `IsGroupIdSet` and move the string comparison in there (checked once and stored in a boolean field).

Reply:
See: https://github.com/BEagle1984/silverback/blob/fa38eb8ef7de7181003ac5efdf2b6bba42be1cb9/src/Silverback.Integration.Kafka/Messaging/Configuration/Kafka/ConfluentClientConfigProxies.cs#L905